terminus/builders/osm_city_builder.py

```diff
@@ -37,35 +37,30 @@ def get_city(self):
         # Get origin of the map (we use the center)
         tree = ET.parse(self.osm_map)
         bounds = tree.find('bounds').attrib
-        self.bounds = {'minlat': float(bounds['minlat']),
-                       'maxlat': float(bounds['maxlat']),
-                       'minlon': float(bounds['minlon']),
-                       'maxlon': float(bounds['maxlon'])}
+        self.bounds = {'origin': LatLon(float(bounds['minlat']), float(bounds['minlon'])),
+                       'corner': LatLon(float(bounds['maxlat']), float(bounds['maxlon']))}
 
-        self.map_origin = LatLon((self.bounds['minlat'] + self.bounds['maxlat']) / 2,
-                                 (self.bounds['minlon'] + self.bounds['maxlon']) / 2)
+        self.map_origin = self.bounds['origin'].middle_point(self.bounds['corner'])
 
         logger.debug("Map Origin: {0}".format(self.map_origin))
 
         # Calculate bounding box based on lat/lon
-        min_xy = self._translate_coords(self.bounds['minlat'],
-                                        self.bounds['minlon'])
-        max_xy = self._translate_coords(self.bounds['maxlat'],
-                                        self.bounds['maxlon'])
+        min_xy = self._translate_coords(self.bounds['origin'])
```

Comment:
Can't we replace all these separate x, y manipulation with points? AFAIK you should be able to do things like `Point(1,2) * 2`